### DISCUSSION TITLE
Chip component with no icon

### DISCUSSION THREAD

**Discussion by puyanwei** [mantinedev/mantine#6763] [Feature requests]
In our designs we have the exact functionality as what is provided by the [Chip component](https://mantine.dev/core/chip/), apart from having no tick/icon.

There doesn't seem to be way of removing the icon, which seems rather strict.

**Comment by rtivital** [comment]
https://codesandbox.io/p/sandbox/mantine-react-template-forked-68vp9x?file=%2Fsrc%2FApp.tsx%3A4%2C14&workspaceId=dd881b63-29f6-464c-bcdf-b5f78ad65b26

**Reply by puyanwei** [reply]
![image](https://github.com/user-attachments/assets/3e4527f4-a03b-42ad-bf46-88a248b8cdda)

It seems a span gets injected into that area (which makes it hard to style) so the icon is removed in your example but the spacing isn't

**Reply by rtivital** [reply]
Chip is designed to work this way to preserve the same width with checked/unchecked state. You can change component padding with Styles API.

**Comment by SteveDowsett** [comment]
Isnt the solution to just not set the icon prop? In the docs there are examples of chips without icons and just text.

**Reply by puyanwei** [reply]
When you have no icon but click on the Chip it will show a blank space where the tick was as shown in the image above

**Comment by SteveDowsett** [comment]
Ah ok, that doesnt seem right, surely that needs to be fixed as its a bug?

**Comment by zobweyt** [comment]
I guess this issue has been fixed in #8382
